replicatedhq/ship
# template entitlement values

What I Did
------------

These wont be queried from the pg graphql-api yet, but this sets us up to start templating them in. Now the template function exists, so ship app mantainers can start coding against the entitlements api

Next step is to expose these on the graphql-api query object and then we can start fetching them.

How I Did it
------------

- add `Entitlements` field to `specs.ShipRelease` and `api.ReleaseMeta`
- Add function to `templates.InstallationContext` that reads entitlement values via e.g.

```
{{repl EntitlementValue "num_seats" }}
```

- I considered adding function `LicenseFieldValue` that mirrors the behavior, to ease migration of replicated YAML to Ship. I don't want to do this just yet, in case we decide to migrate/port License Field Values to entitlments in a different way.  This is definitely something to consider, but is something that is easier to add in later than it is to take out once we've added it.
- add `--set-entitlements-json` flag for local "studio" dev and integration testing

How to verify it
------------

```
make build && \
bin/ship app \
    --runbook integration/base/entitlements-local/input/.ship/release.yml \
    --set-entitlements-json '{"values":[{"key":"num_seats","value":"100}]}' \
    --headless
```

Verify the generate script uses your value

```
$ sh installer/scripts/seats.sh
you get 100 seats
```

Description for the Changelog
------------

Add preliminary support for templating Replicated entilements values via `{{repl EntitlementValue "NAME"}}`

:boat:
